src/Client.php

```
@@ -275,4 +275,151 @@ public function getAsset($assetId) {
     return (string) $response->getBody();
   }
 
+  /**
+   * Gets presigned url from AWS S3.
+   *
+   * @param array $file_data
+   *   The file data required by Webdam.
+   *
+   * @return mixed
+   *   Presigned url needed for next step + PID.
+   */
+  public function getPresignUrl(array $file_data) {
+    $this->checkAuth();
+
+    try {
+      $response = $this->client->request(
+        "GET",
+        $this->baseUrl . '/ws/awss3/generateupload',
+        [
+          'headers' => $this->getDefaultHeaders(),
+          'query' => $file_data,
+        ]
+      );
+      return json_decode($response->getBody());
+    }
+    catch (ClientException $e) {
+      dump($e->getMessage());
+      return FALSE;
```

Just let the Exception bubble to the caller, and only return valid data.